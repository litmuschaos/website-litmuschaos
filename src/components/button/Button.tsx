import styled from "styled-components";

interface IButton {
  gradientColor?: "purple" | "green";
  backgroundColor?: string;
}

interface IOutlinedButton {
  backgroundColor: string;
  width?: string;
}

// Default Button Styles
const Button = styled.button<IButton>`
  height: 3rem;
  min-width: 10rem;
  width: ${(props) => (props.theme.screens.sm ? "100%" : "18rem")};
  border: none;
  background: ${(props) =>
    props.gradientColor === "purple"
      ? props.theme.gradient.purple
      : props.gradientColor === "green"
      ? props.theme.gradient.green
      : props.backgroundColor};
  color: white;
  border-radius: 0.25rem;
  font-size: ${(props) => props.theme.fontSize.button};
  cursor: pointer;
  :disabled {
    background: lightgray;
  }
`;

// White On Green Button Styles
const WhiteOnGreenButton = styled.button`
  height: 2rem;
  min-width: 13rem;
  border: none;
  border-radius: 0.2rem;
  background: white;
  color: ${(props) => props.theme.colors.darkGreen(1)};
  font-size: ${(props) => props.theme.fontSize.small.lg};
  font-weight: bold;
  cursor: pointer;
  box-shadow: 0px 5px 13px rgba(0, 0, 0, 0.08);
  :disabled {
    background: lightgray;
  }
`;

// Outline Button Styles
const OutlinedButton = styled.button<IOutlinedButton>`
  min-width: 2rem;
  padding: 0.5rem 2rem;
  background: transparent;
  width: ${(props) => (props.theme.screens.sm ? "100%" : "max-content")};
  color: ${(props) => props.backgroundColor};
  border: 0.05rem solid ${(props) => props.backgroundColor};
  border-radius: 0.5rem;
  font-size: ${(props) => props.theme.fontSize.button};
  cursor: pointer;
  :disabled {
    background: lightgray;
  }
`;

export { Button, OutlinedButton, WhiteOnGreenButton };
