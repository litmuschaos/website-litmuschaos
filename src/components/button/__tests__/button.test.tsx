import { render } from "@testing-library/react";
import React from "react";
import { Button } from "../Button";

it("matches snapshot", () => {
  const { asFragment } = render(<Button>Hello World!</Button>);
  expect(asFragment()).toMatchSnapshot();
});
