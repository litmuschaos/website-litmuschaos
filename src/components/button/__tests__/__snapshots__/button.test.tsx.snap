// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa cjEIaC"
  >
    Hello World!
  </button>
</DocumentFragment>
`;
